web-module/src/main/webapp/resources/structEditJS/StructEditor.js

```diff
@@ -0,0 +1,801 @@
+            elemValue = fieldElem.getValue();
+          } else {
+            return false;
+          }
+        } else if (isInputElem && (fieldElem.type.toLowerCase() == 'checkbox')) {
+          elemValue = fieldElem.checked;
+        }
+        return (_me._initialValues.get(fieldElem.name) != elemValue);
+      }
+      return false;
+    },
+
+    isDirty : function() {
+      var _me = this;
+      var isDirty = false;
+      if (_me.isValidFormId()) {
+        if (_me._formDirtyOnLoad()) {
+          console.log('isDirty formDirtyOnLoad found. ');
+          isDirty = true;
+        } else {
+          _me._updateTinyMCETextAreas();
+          _me._formElem.getElements().each(function(elem) {
+            if (_me._isSubmittableField(elem) && _me._isDirtyField(elem)) {
+              console.log('isDirty first found dirty field: ', elem.name);
+              isDirty = true;
+              throw $break;  //prototype each -> break
+            }
+          });
+        }
+      } else {
+        console.warn('isDirty: form with id [' + _me.formElem.id
+            + '] disappeared since loading the editor.');
+      }
+      return isDirty;
+    }
+
+  });
+
+  /************************************
+   * CelementsFormSaver class definition *
+   ************************************/
+  window.CELEMENTS.structEdit.CelementsFormSaver = Class.create({
+    _jsonResponses : undefined,
+    _saveCallback : undefined,
+
+    initialize : function(saveCallback) {
+      var _me = this;
+      _me._jsonResponses = new Hash();
+      _me._saveCallback = saveCallback;
+    },
+
+    _saveAndContinueAjax : function(formName, handler) {
+      var _me = this;
+      if(document.forms[formName]) {
+        document.forms[formName].select('textarea.mceEditor').each(function(formfield) {
+           console.log('textarea save tinymce: ', formfield.name, tinyMCE.get(formfield.id).save());
+           formfield.value = tinyMCE.get(formfield.id).save();
+         });
+         $(formName).request(handler);
+       } else {
+         console.error('form not found: ', formName);
+       }
+     },
+
+     _handleSaveAjaxResponse : function(formId, transport) {
+       var _me = this;
+       if (transport.responseText.isJSON()) {
+         console.log('_handleSaveAjaxResponse with json result: ', transport.responseText);
+         var jsonResult = transport.responseText.evalJSON();
+         _me._jsonResponses.set(formId, jsonResult);
+         if (jsonResult.successful) {
+           return true;
+         } else {
+           console.warn('_handleSaveAjaxResponse: save failed for [' + formId + ']: ', jsonResult);
+         }
+       } else {
+         return true;
+       }
+       return false;
+     },
+
+    saveAllForms : function(allDirtyFormIds) {
+      var _me = this;
+      if (allDirtyFormIds.size() > 0) {
+        var formId = allDirtyFormIds.pop();
+        var remainingDirtyFormIds = allDirtyFormIds;
+        _me._saveAndContinueAjax(formId, {
```

Comment:
Either use onComplete and handle problems or add onFailed(?) ... otherwiese e.b. Velocity-Exceptions could lead to wrong behaviours